lighthouse-core/driver/gatherers/interstitial.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gather = require('./gather');
+
+/* global window, document, __returnResults */
+
+/* istanbul ignore next */
+function getInterstitial() {
+  const viewportSize = window.outerWidth * window.outerHeight;
+
+  const hasNonTransparentBackground = computedStyle => {
+    // Simple early exit for a solid background color.
+    const bgColor = computedStyle.backgroundColor;
+    const hasSolidBackground = (/#\d{6}/.test(bgColor) ||
+        /rgb\(/.test(bgColor) ||
+        /hsl\(/.test(bgColor) ||
+        /^\w+$/.test(bgColor));
+
+    if (hasSolidBackground) {
+      return true;
+    }
+
+    // Next try rgba.
+    const hasRGBA = /^rgba\([^\)]+\)/.exec(bgColor);
+    if (hasRGBA !== null && hasRGBA.length >= 2) {
+      const alpha = parseFloat(hasRGBA[1].split(',').slice(-1));
+      if (alpha > 0) {
+        return true;
+      }
+    }
+
+    // Then HSLA.
+    const hasHSLA = /^hsla\([^\)]+\)/.exec(bgColor);
+    if (hasHSLA !== null && hasHSLA.length >= 2) {
+      const alpha = parseFloat(hasHSLA[1].split(',').slice(-1));
+      if (alpha > 0) {
+        return true;
+      }
+    }
+
+    // Finally try and see if there's a background image.
+    const hasBackgroundImage = computedStyle.backgroundImage !== 'none';
+    const backgroundRepeats = /repeat/.test(computedStyle.backgroundRepeat);
+
+    return (hasBackgroundImage && backgroundRepeats);
+  };
+
+  // Walk the tree of elements...
+  const candidates = [...document.querySelectorAll('*')]
+    .filter(e => {
+      // Check for position fixed.
+      const computedStyle = window.getComputedStyle(e);
+      const isFixedOrAbsolute = computedStyle.position === 'fixed' ||
+          computedStyle.position === 'absolute';
```

Comment:
how much of this check could be broken down into individual steps and done via the debugging protocol? A little more roundabout, but makes things like this a bit easier and testing could be cleaner